autogpt_platform/backend/backend/server/routers/integrations.py

```diff
@@ -80,20 +81,45 @@ async def callback(
     user_id: Annotated[str, Depends(get_user_id)],
     request: Request,
 ) -> CredentialsMetaResponse:
+    logger.debug(f"Received OAuth callback for provider: {provider}")
     handler = _get_provider_oauth_handler(request, provider)
 
     # Verify the state token
     if not await store.verify_state_token(user_id, state_token, provider):
+        logger.warning(f"Invalid or expired state token for user {user_id}")
         raise HTTPException(status_code=400, detail="Invalid or expired state token")
 
     try:
-        credentials = handler.exchange_code_for_tokens(code)
+        scopes = await store.get_scopes_from_state_token(user_id, state_token, provider)
+        logger.debug(f"Retrieved scopes from state token: {scopes}")
+
+        # If scopes are empty, use the default scopes for the provider
+        if not scopes:
+            logger.debug(f"Using default scopes for provider {provider}")
+            scopes = handler.DEFAULT_SCOPES
```

Comment:
This should be in the `BaseOAuthHandler` IMO

Reply:
done